src/wavesurfer.js

```diff
@@ -232,6 +233,18 @@ const WaveSurfer = util.extend({}, util.observer, { util }, {
         this.drawer = Object.create(this.Drawer);
         this.drawer.init(this.container, this.params);
         this.fireEvent('drawer-created', this.drawer);
+        let prevWidth = 0;
+        this._onResize = () => {
+            if (prevWidth != this.drawer.wrapper.clientWidth) {
+                prevWidth = this.drawer.wrapper.clientWidth;
+                this.empty();
+                this.drawBuffer();
+            }
+        };
+
+        if (this.params.responsive) {
+            window.addEventListener('resize', this._onResize, true);
+        }
```

Comment:
@mspae I would do the following instead to avoid running drawBuffer() continuously on resize event. It will hang the browser or crash it depending on how big the waveform is. 100 - 200ms should be fine, I set the timeout to 150ms.

````
let prevWidth = 0;
var resp;

this._onResize = () => {
   clearTimeout(resp)
   resp = setTimeout(function(){
      if (prevWidth != this.drawer.wrapper.clientWidth) {
                prevWidth = this.drawer.wrapper.clientWidth;
                this.empty();
                this.drawBuffer();
      }
   }, 150);
};

if (this.params.responsive) {
    window.addEventListener('resize', this._onResize, true);
}
````